reco_utils/dataset/pandas_df_utils.py

```diff
@@ -166,3 +167,97 @@ def _convert(field, feature, field_index, field_feature_index_dict):
     return df_new
 
 
+def negative_feedback_sampler(
+    df, 
+    col_user=DEFAULT_USER_COL,
+    col_item=DEFAULT_ITEM_COL,
+    col_label=DEFAULT_LABEL_COL,
+    number_neg_per_user=1,
```

Comment:
Are there any cases we want to have imbalanced labels? If not, why don't we just remove `number_neg_per_user` argument and make this function to (try to) sample the same number of negative examples as positive's?

If we still want to have this option, wouldn't be better to pass the ratio to positive samples? Similar to `pd.DataFrame.sample`'s fraction argument.

Reply:
Good point. `ratio_neg_per_user` sounds a good idea.

Reply:
Updated.